Add action on matrices in row reduced echelon form

ThomasBreuer opened the pull request:
This implements the action of a matrix group on subspaces of its natural module.

ThomasBreuer committed:
add action on matrices in row reduced echelon form

This implements the action of a matrix group on subspaces
of its natural module.

thofma commented on src/Groups/action.jl:
````diff
@@ -453,6 +454,45 @@ end
+```
+"""
+function on_rref_mats(m::MatElem{T}, x::MatrixGroupElem) where T <: FinFieldElem
+  return rref(m * x)[2]
````
```suggestion
  return echelon_form(m * x)
```

thofma replied:
It is the same, but since we have `echelon_form`, we might as well make use of it.

ThomasBreuer committed:
address comments

Call `echelon_form` instead of `rref`,
rename the function accordingly.
(Also `echelon_form` cannot be referenced via `@ref`.)

ThomasBreuer committed:
fix the return value

(`rref` and `echelon_form` have different output ...)

mjrodgers reviewed (approved):
This looks good. It seems to be a good starting point for what I need, thanks Thomas..
